### PR TITLE
IN-2492 Add lost_reason to deals.info

### PR DESCRIPTION
**Pull request by luisaterrabrito**

### CODE REVIEW

**Review by sanderbrugge — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by steftrenson — DISMISSED** [review]

**Review by tiagobastos — DISMISSED** [review]

File: src/03-deals/deals.apib [hunk 1]
```diff
@@ -157,6 +157,11 @@ Get details for a single deal.
             + updated_at: `2017-05-09T11:30:58+00:00` (string)
             + `web_url`: `https://app.teamleader.eu/sale_detail.php?id=f6871b06-6513-4750-b5e6-ff3503b5a029` (string)
             + custom_fields (array[CustomField])
+            + lost_reason (object)
+                + reason (object)
+                    + type: lost_reason (string)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: B: types are camel cased, so this should be `lostReason`.

IIRC this is also a requirement to have the lost reasons sideloaded.

File: src/03-deals/deals.apib [hunk 2]
```diff
@@ -157,6 +157,11 @@ Get details for a single deal.
             + updated_at: `2017-05-09T11:30:58+00:00` (string)
             + `web_url`: `https://app.teamleader.eu/sale_detail.php?id=f6871b06-6513-4750-b5e6-ff3503b5a029` (string)
             + custom_fields (array[CustomField])
+            + lost_reason (object)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: PP: place the `lost_reason` _before_ meta data properties (`created_at`, `updated_at` and `web_url`).

File: src/changes-backwards-compatible.apib [hunk 3]
```diff
@@ -6,6 +6,8 @@ We list all backwards-compatible additions here. These are currently available i
 
 #### Latest
 
+- We added `lost_reason` to the response of `deals.info`.
```
Comment (dietrichm) [inline src/changes-backwards-compatible.apib:9]: PP: have #494 as the base branch and add `deals.info` to the addition added there.

**Review by dietrichm — CHANGES_REQUESTED** [review]

File: apiary.apib [hunk 4]
```diff
@@ -1748,6 +1750,11 @@ Get details for a single deal.
             + updated_at: `2017-05-09T11:30:58+00:00` (string)
             + `web_url`: `https://app.teamleader.eu/sale_detail.php?id=f6871b06-6513-4750-b5e6-ff3503b5a029` (string)
             + custom_fields (array[CustomField])
+            + lost_reason (object)
+                + reason (object)
+                    + type: lost_reason (string)
```
Comment (sanderbrugge) [inline apiary.apib]: Q: I guess this should be camelcased too?

**Review by tiagobastos — DISMISSED** [review]

**Review by sanderbrugge — DISMISSED** [review]

**Review by dietrichm — DISMISSED** [review]

**Review by EOSullivanBerlin — DISMISSED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Review by tiagobastos — APPROVED** [review]

File: src/03-deals/deals.apib [hunk 5]
```diff
@@ -153,11 +153,11 @@ Get details for a single deal.
...
                     + id: `e2314517-3cab-4aa9-8471-450e73449041` (string)
                     + type: `quotation` (string)
-            + lost_reason (object)
-                + reason (object)
+            + lost_reason (object, nullable)
+                + reason (object, nullable)
                     + type: lostReason (string)
                     + id: `4e81f3c4-7dca-44cb-b126-6158bec392a2` (string)
-                + remark: `Too expensive` (string)
+                + remark: `Too expensive` (string, nullable)
```
Comment (dietrichm) [inline src/03-deals/deals.apib]: Q: can it happen that `lost_reason` is provided, but both or either reason or remark are `null`?

Comment (dietrichm) [reply src/03-deals/deals.apib]: Explained by @luisaterrabrito IRL; if the reason is unknown, there is no `lostReason` set.

**Review by dietrichm — APPROVED** [review]

**Review by sanderbrugge — APPROVED** [review]

**Review by EOSullivanBerlin — APPROVED** [review]

**Pull request merged**